src/main/flight/failsafe.c

```diff
@@ -93,6 +95,16 @@ void failsafeInit(void)
     failsafeState.events = 0;
     failsafeState.monitoring = false;
 
+    // Disallow failsafe procedure GPS-RESCUE if 3D mode is active or GPS Rescue is not available
+    if (failsafeConfig()->failsafe_procedure == FAILSAFE_PROCEDURE_GPS_RESCUE && (
+        feature(FEATURE_3D)
+#ifndef USE_GPS_RESCUE
+        || true
+#endif
+        )) {
+        failsafeConfigMutable()->failsafe_procedure = FAILSAFE_PROCEDURE_DROP_IT;
```

Comment:
It should be a part of `validateConfig` IMO.

Reply:
I knew someone was going to say that...  :)  I originally had it there but the problem is that saving from the Configuration tab in the configurator never calls `validateConfig`.  So it's possible to enable the 3D feature but still leave the `failsafe_procedure` set to GPS-RESCUE.